packages/react/src/Button/IconButton.tsx

```diff
@@ -4,20 +4,62 @@ import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/po
 import {ButtonBase} from './ButtonBase'
 import {defaultSxProp} from '../utils/defaultSxProp'
 import {generateCustomSxProp} from './Button'
+import {Tooltip} from '../drafts/Tooltip/Tooltip'
 
-const IconButton = forwardRef(({sx: sxProp = defaultSxProp, icon: Icon, ...props}, forwardedRef): JSX.Element => {
-  let sxStyles = sxProp
-  // grap the button props that have associated data attributes in the styles
-  const {size} = props
+const IconButton = forwardRef(
+  (
+    {
+      sx: sxProp = defaultSxProp,
+      icon: Icon,
+      'aria-label': ariaLabel,
+      description,
+      disabled,
+      // This is planned to be a temporary prop until the default tooltip on icon buttons are fully rolled out.
+      disableTooltip = false,
+      ...props
+    },
+    forwardedRef,
+  ): JSX.Element => {
+    let sxStyles = sxProp
+    // grap the button props that have associated data attributes in the styles
+    const {size} = props
 
-  if (sxProp !== null && Object.keys(sxProp).length > 0) {
-    sxStyles = generateCustomSxProp({size}, sxProp)
-  }
+    if (sxProp !== null && Object.keys(sxProp).length > 0) {
+      sxStyles = generateCustomSxProp({size}, sxProp)
+    }
 
-  return (
-    // @ts-expect-error StyledButton wants both Anchor and Button refs
-    <ButtonBase icon={Icon} data-component="IconButton" sx={sxStyles} type="button" {...props} ref={forwardedRef} />
-  )
-}) as PolymorphicForwardRefComponent<'button' | 'a', IconButtonProps>
+    const withoutTooltip = disableTooltip || disabled || ariaLabel === undefined || ariaLabel === ''
```

Comment:
Since we have the ability to keep the tooltip off on icon buttons, I was wondering if we still need to have a workaround for the cases where icon buttons that are already wrapped with Tooltip 🤔

```jsx
<Tooltip text="Notifications" type="label" >
  <IconButton icon={X} aria-label="Notifications" />
</Tooltip>
```

Above usage, is discourage by our accessibility checks both in [dotcom axe rules](https://github.com/github/axe-github/blob/main/src/index.ts#L137) and in [PRC Tooltip v2 implementation](https://github.com/primer/react/blob/main/packages/react/src/drafts/Tooltip/Tooltip.tsx#L228). 

I am thinking not to cover this case in the implementation because it is not a usage we encourage. I would love to hear your thoughts on this 🙌🏻